### PR TITLE
refactoring requesting methods

### PR DESCRIPTION
**Pull request by robechon**
Signed-off-by: Robin BECHON <robin@cyberwatch.fr>

### CODE REVIEW

File: cbw_api_toolbox/cbw_api.py [hunk 1]
```diff
@@ -26,13 +26,15 @@ def __init__(self, api_url, api_key, secret_key, verify_ssl=False):
     @staticmethod
     def _get_params(params=None):
         return "/{0}".format('/'.join(params)) if params else ""
-
-    def _request(self, route, params=None):
+
```
Comment (fakenine) [inline cbw_api_toolbox/cbw_api.py]: There are blank spaces here

File: cbw_api_toolbox/cbw_api.py [hunk 2]
```diff
@@ -45,15 +47,23 @@ def _request(self, route, params=None):
             self.logger.error("An error occurred, please check your API_URL.")
             os._exit(-1)
 
+    # GET request to /api/v2/ping then check uuid value
```
Comment (fakenine) [inline cbw_api_toolbox/cbw_api.py]: The comment should be right above the method definition

File: cbw_api_toolbox/cbw_api.py [hunk 3]
```diff
@@ -45,15 +47,23 @@ def _request(self, route, params=None):
...
+
     def ping(self):
-        result = self._request(ROUTE_GET_PING)
+        result = self._request(ROUTE_PING, "GET")
         print("OK") if result and 'uuid' in result else self.logger.error("FAILED")
 
+    # GET request to /api/v2/servers to get all servers
```
Comment (fakenine) [inline cbw_api_toolbox/cbw_api.py]: The comment should be right above the method definition

File: cbw_api_toolbox/cbw_api.py [hunk 4]
```diff
@@ -45,15 +47,23 @@ def _request(self, route, params=None):
...
+
     def servers(self):
-        return [CBWParser().parse(CBWServer, server) for server in self._request(ROUTE_GET_SERVERS)]
+        return [CBWParser().parse(CBWServer, server) for server in self._request(ROUTE_SERVERS, "GET")]
+
+    # GET request to /api/v2/servers to get all informations about a specific server
```
Comment (fakenine) [inline cbw_api_toolbox/cbw_api.py]: The comment should be right above the method definition

File: cbw_api_toolbox/cbw_api.py [hunk 5]
```diff
@@ -45,15 +47,23 @@ def _request(self, route, params=None):
...
 
     def server(self, server_id):
-        return CBWParser().parse(CBWServer, self._request(ROUTE_GET_SERVERS, [server_id]))
+        return CBWParser().parse(CBWServer, self._request(ROUTE_SERVERS, "GET", [server_id]))
+
+    # Use servers method to get all informations for each server
```
Comment (fakenine) [inline cbw_api_toolbox/cbw_api.py]: The comment should be right above the method definition

File: cbw_api_toolbox/cbw_api.py [hunk 6]
```diff
@@ -26,13 +26,15 @@ def __init__(self, api_url, api_key, secret_key, verify_ssl=False):
...
-
-    def _request(self, route, params=None):
+    
+    def _request(self, route, verb, params=None):
         raw_response = None
         try:
-            raw_response = requests.get("{0}{1}{2}".format(self.api_url, route, self._get_params(params)),
-                                        auth=CBWAuth(self.api_key, self.secret_key),
-                                        verify=self.verify_ssl)
+            url = self.api_url + route + self._get_params(params)
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: Please use format in place of concatenation.

https://softwareengineering.stackexchange.com/questions/304445/why-is-s-better-than-for-concatenation

Comment (fwininger) [reply cbw_api_toolbox/cbw_api.py]: Please don't assign attribute like `verify = self.verify_ssl`

```python
raw_response = requests.request(verb,
                                "{0}{1}{2}".format(self.api_url, route, self._get_params(params)),
                                auth=CBWAuth(self.api_key, self.secret_key),
                                verify=self.verify_ssl)
```

File: cbw_api_toolbox/cbw_api.py (hunk 3 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

File: cbw_api_toolbox/cbw_api.py (hunk 4 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

File: cbw_api_toolbox/cbw_api.py (hunk 5 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

File: cbw_api_toolbox/cbw_api.py (hunk 6 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

File: cbw_api_toolbox/cbw_api.py [hunk 7]
```diff
@@ -8,12 +8,11 @@
 from urllib3.exceptions import NewConnectionError, MaxRetryError
 
 from cbw_api_toolbox import API_DEFAULT_URL
-from cbw_api_toolbox.__routes__ import ROUTE_GET_SERVERS, ROUTE_GET_PING
+from cbw_api_toolbox.__routes__ import ROUTE_SERVERS, ROUTE_PING
 from cbw_api_toolbox.cbw_auth import CBWAuth
 from cbw_api_toolbox.cbw_objects.cbw_server import CBWServer
 from cbw_api_toolbox.cbw_parser import CBWParser
 
-
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: https://www.python.org/dev/peps/pep-0008/#blank-lines

File: cbw_api_toolbox/cbw_api.py [hunk 8]
```diff
@@ -45,15 +46,19 @@ def _request(self, route, params=None):
...
 
+    # GET request to /api/v2/servers to get all informations about a specific server
     def server(self, server_id):
-        return CBWParser().parse(CBWServer, self._request(ROUTE_GET_SERVERS, [server_id]))
+        return CBWParser().parse(CBWServer, self._request(ROUTE_SERVERS, "GET", [server_id]))
 
+    # Use servers method to get all informations for each server
     def get_detailed_servers(self):
-        return [self.server(server.server_id) for server in self.servers()]
+        return [self.server(server.server_id) for server in self.servers()]
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: No new line

File: cbw_api_toolbox/cbw_api.py [hunk 9]
```diff
@@ -27,12 +26,14 @@ def __init__(self, api_url, api_key, secret_key, verify_ssl=False):
...
-    def _request(self, route, params=None):
+    def _request(self, route, verb, params=None):
         raw_response = None
+        
         try:
-            raw_response = requests.get("{0}{1}{2}".format(self.api_url, route, self._get_params(params)),
-                                        auth=CBWAuth(self.api_key, self.secret_key),
-                                        verify=self.verify_ssl)
+            raw_response = requests.request(verb, "{0}{1}{2}".format(self.api_url, route, self._get_params(params)),
+                            data=params,
```
Comment (fwininger) [inline cbw_api_toolbox/cbw_api.py]: https://www.python.org/dev/peps/pep-0008/#indentation

File: cbw_api_toolbox/cbw_api.py (hunk 7 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

File: cbw_api_toolbox/cbw_api.py (hunk 9 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

File: cbw_api_toolbox/cbw_api.py (hunk 8 above)
Comment (robechon) [reply cbw_api_toolbox/cbw_api.py]: Done

**Pull request merged**